### PR TITLE
infra: replace nginx with apisix

### PR DESCRIPTION
**Pull request by MaxSchaefer**

### CODE REVIEW

**Comment by MaxSchaefer** [comment]
@FoseFx does this pr closes #58?

**Comment by MaxSchaefer** [comment]
> Is there any good way to test this against a local cluster?

You can install APISIX via Helm into your local Dapr-ready cluster [by following this guide step one through three](https://apisix.apache.org/blog/2021/11/17/dapr-with-apisix/#step-1-apache-apisix-helm-configuration).

**Comment by FoseFx** [comment]
So I have to copy the values? I hoped there would be a way to run terraform against kind or sth. :/

**Comment by FoseFx** [comment]
Is this ready for review?

**Comment by MaxSchaefer** [comment]
> So I have to copy the values? I hoped there would be a way to run terraform against kind or sth. :/

I've developed the `local`-Environment. You can now apply `/kubernetes/iac/environemnts/local` with terraform into your cluster of choice.

File: kubernetes/iac/modules/hw-tf-module-cluster-resources/main.tf [hunk 1]
```diff
@@ -0,0 +1,102 @@
+resource "helm_release" "dapr" {
```
Comment (FoseFx) [inline kubernetes/iac/modules/hw-tf-module-cluster-resources/main.tf:1]: Can we split this into multiple files? One for each release. I don't think we can keep extending the one .tf file especially when we want to migrate all the helmfile configs to terraform.

File: kubernetes/iac/environments/local/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# iac-environments-local
+
+Apply the cluster-resources module to your local development cluster
+
+### Port-Forward the APISIX Dashboard or Gateway
+> kubectl port-forward [name of the apisix-dashboard-pod] 9000:9000
```
Comment (FoseFx) [inline kubernetes/iac/environments/local/README.md:6]: Forwarding to the services is probably better practice.

File: kubernetes/iac/environments/local/README.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+
+> kubectl port-forward [name of the apisix-gateway-pod] 9080:9080
```
Comment (FoseFx) [inline kubernetes/iac/environments/local/README.md:8]: Can you add a section that describes that user/password is admin/admin? Also please add a follow up issue to use a newly generated k8s secret instead.

**Review by FoseFx — APPROVED** [review]
Nice work! I added some minor change requests. Can you also create a follow-up issue for APISIX CRDs after merging?

File: kubernetes/iac/modules/hw-tf-module-cluster-resources/main.tf [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  }
+}
+
+resource "helm_release" "apisix" {
+  name  = "apisix"
+  repository = "https://charts.apiseven.com"
+  chart = "apisix"
+
+  depends_on = [
+    helm_release.dapr
```
[the remark this answers is not in the document]
Comment (MaxSchaefer) [reply kubernetes/iac/modules/hw-tf-module-cluster-resources/main.tf]: Unfortunately not. APISIX depends on Dapr for the sidecar mechanism.

**Pull request merged**